frontend/catalyst/from_plxpr/from_plxpr.py

```diff
@@ -240,6 +273,74 @@ def calling_convention(*args):
 }
 
 
+def apply_compiler_decompose_to_plxpr(inner_jaxpr, consts, tgateset, ncargs):
+    """Apply the compiler-specific decomposition for a given JAXPR.
+
+    Args:
+        inner_jaxpr (Jaxpr): The input JAXPR to be decomposed.
+        consts (list): The constants used in the JAXPR.
+        tgateset (list): A list of target gateset for decomposition.
+        ncargs (list): Non-constant arguments for the JAXPR.
+        qargs (list): All arguments including constants and non-constants.
+
+    Returns:
+        ClosedJaxpr: The decomposed JAXPR.
+    """
+
+    # Disable the graph decomposition optimization
+
+    # Why? Because for the compiler-specific decomposition we want to
+    # only decompose higher-level gates and templates that only have
+    # a single decomposition, and not do any further optimization
+    # based on the graph solution.
+    # Besides, the graph-based decomposition is not supported
+    # yet in from_plxpr for most gates and templates.
+
+    # TODO: Enable the graph-based decomposition
+    qml.decomposition.disable_graph()
+
+    # First perform the pre-mlir decomposition to simplify the jaxpr
+    # by decomposing high-level gates and templates
+    gate_set = COMPILER_OPERATIONS + tgateset
+
+    final_jaxpr = qml.transforms.decompose.plxpr_transform(
+        inner_jaxpr, consts, (), {"gate_set": gate_set}, *ncargs
+    )
+
+    qml.decomposition.enable_graph()
+
+    return final_jaxpr
+
+
+def collect_and_compile_graph_solutions(inner_jaxpr, consts, tkwargs, ncargs):
+    """Collect and compile graph solutions for a given JAXPR.
+
+    This function uses the DecompRuleInterpreter to evaluate
+    the input JAXPR and obtain a new JAXPR that incorporates
+    the graph-based decomposition solutions.
+
+    This function doesn't modify the underlying quantum function
+    but rather constructs a new JAXPR with decomposition rules.
```

Comment:
I'm slightly confused. I thought the `DecompRuleInterpreter` only collects information and builds a graph? How are decomposition rules incorporated into the JAXPR?